### DISCUSSION TITLE
Client global search function

### DISCUSSION THREAD

**Discussion by Rocky-Jinxing** [rustdesk/rustdesk-server-pro#377] [Feature Request]
The client searches according to the address book, and cannot search for devices in other address books. We have added more than 100 controlled devices, and there will be more and more in the future. This search brings us great inconvenience. We hope to add a global search function.

**Comment by rustdesk** [comment]
Will be considered, but low priority and need to get more suggesion  from other users.

**Comment by rustdesk** [comment]
Do you mean on RustDesk client or on web console?

**Reply by Rocky-Jinxing** [reply]
> Do you mean on RustDesk client or on web console?

@rustdesk RustDesk client address book global search, as it can only search in each address book now (I told you we already have 231 controlled endpoints, so sometimes it's very difficult to find them)

**Reply by Mhalkyo** [reply]
> Voulez-vous dire sur le client RustDesk ou sur la console Web ?

I speak into the console

**Reply by Mhalkyo** [reply]
Here is an example. without necessarily having the number of positions and if they are connected but to search in all groups at the same time on the console
![image](https://github.com/user-attachments/assets/a1f11850-00a7-47d5-9da3-29d763d6aa1a)

**Comment by ResidentViking** [comment]
Yes, implement a global search please in the RustDesk client's address book. This could be done by adding an "All Devices" option above "My Address Book," or by making the search bar default to searching the entire address book when a name is entered.

**Comment by rustdesk** [comment]
Is it on rustdesk client or web console?

**Reply by Rocky-Jinxing** [reply]
> rustdesk client

rustdesk client

By default, an address book will be automatically selected, and the search can only search the hosts in this address book, and cannot search the hosts in the entire system.

**Comment by rustdesk** [comment]
@21pages

**Comment by jolveira-DF** [comment]
We really like RustDesk and use it daily as a Pro user.

In the RustDesk client search only works within the selected address book, not across all address books.
Please consider adding a global search for all controlled devices, such as:
 a) An “All Devices” option above “My Address Book”, or
 b) Making the search bar search all address books by default.
With 400+ devices this makes finding endpoints inefficient. We often have to search for a client in one place and then search notes or related information elsewhere.
This would significantly improve usability.

Thank you.

**Comment by JMBranco64** [comment]
If you have hundreds of devices, why do you have many address books. I am starting with RustDesk and organizing to have 1 shared address book amongst all tecs.

**Comment by StealUrKill** [comment]
I would like a shared search for all as well. We don't use address book just devices groups that show up under Accessible Devices tab. Also a way to expand the side column would be a great benefit as well. For option 1 the ideal option would to be able to search all devices and for option 2 is the ability to expand the list section.

I understand if you do not have have any devices groups selected you can search all but sometimes they forget to uncheck. searching by hostname, user, or even id would be swell.

<img width="617" height="203" alt="image" src="https://github.com/user-attachments/assets/1e8549c9-d007-4117-8d8a-cb631fd4cc83" />
